Userland/Libraries/LibJS/Runtime/DateConstructor.cpp

```diff
@@ -182,6 +182,7 @@ static double parse_date_string(VM& vm, ByteString const& date_string)
         "%a%t%b%t%e%t%Y%t%T%t%Z"sv,            // "Tue Nov 07 2023 10:05:55  UTC"
         "%a%t%b%t%e%t%T%t%Y"sv,                // "Wed Apr 17 23:08:53 2019"
         "%a%t%b%t%e%t%Y%t%T"sv,                // "Wed Apr 17 2019 23:08:53"
+        "%Y-%m-%eT%T%z"sv,                     // "2024-10-19T11:22:00+0000"
```

Comment:
Isn't this valid iso8601 already? I thought we'd parse that already. Why can't the existing parsing code handle the date in the comment?

Reply:
Apparently it was not supported before, there was a `Unable to parse date string` log entry in the terminal when visiting some subpage of `rainews.it`
```
16722.288 WebContent(17227): Unable to parse date string: "2024-10-27T02:11:00+0000"
16722.288 WebContent(17227): Unable to parse date string: "2024-10-26T11:45:00+0000"
16722.288 WebContent(17227): Unable to parse date string: "2024-10-26T11:43:00+0000"
16722.288 WebContent(17227): Unable to parse date string: "2024-10-26T17:19:00+0000"
16722.288 WebContent(17227): Unable to parse date string: "2024-10-27T14:22:00+0000"
16722.288 WebContent(17227): Unable to parse date string: "2024-10-26T11:26:00+0000"
16722.288 WebContent(17227): Unable to parse date string: "2024-10-27T14:43:00+0000"
16722.288 WebContent(17227): Unable to parse date string: "2024-10-27T13:49:00+0000"
16722.288 WebContent(17227): Unable to parse date string: "2024-10-19T11:22:00+0000"
16722.288 WebContent(17227): Unable to parse date string: "2024-09-30T14:24:00+0000"
16722.288 WebContent(17227): Unable to parse date string: "2024-09-30T14:32:00+0000"
16722.288 WebContent(17227): Unable to parse date string: "2024-10-26T17:51:00+0000"
```

Reply:
The iso8601 format requires a `:` between the last `+0000`